File: win-prog/ProgWinCS/Text and Fonts/FontSizes/FontSizes.cs
//----------------------------------------
// FontSizes.cs © 2001 by Charles Petzold
//----------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class FontSizes: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new FontSizes());
     }
     public FontSizes()
     {
          Text = "Font Sizes";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {    
          string strFont = "Times New Roman";
          Brush  brush   = new SolidBrush(clr);
          float  y       = 0;

          for (float fSize = 6; fSize <= 12; fSize += 0.25f)
          {
               Font font = new Font(strFont, fSize);
               grfx.DrawString(strFont + " in " + fSize + " points", 
                               font, brush, 0, y);
               y += font.GetHeight(grfx);
          }
     }
}

File: win-prog/ProgWinVB/Text and Fonts/FontSizes/FontSizes.vb
'------------------------------------------
' FontSizes.vb (c) 2002 by Charles Petzold
'------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class FontSizes
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(new FontSizes())
    End Sub

    Sub New()
        Text = "Font Sizes"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim strFont As String = "Times New Roman"
        Dim br As New SolidBrush(clr)
        Dim y As Single = 0
        Dim fSize As Single

        For fSize = 6 To 12 Step 0.25
            Dim fnt As New Font(strFont, fSize)
            grfx.DrawString(strFont & " in " & Str(fSize) & " points", _
                            fnt, br, 0, y)
            y += fnt.GetHeight(grfx)
        Next fSize
    End Sub
End Class
